src/panes/ManualInput.jsx

```diff
@@ -0,0 +1,220 @@
+    onValueChange(isNaN(newValue) ? 0 : newValue)
+  }
+
+  const increment = (step) => () => {
+    setInputValue((Math.max(min, Math.min(max, value + step))).toString())
+    onValueChange(Math.max(min, Math.min(max, value + step)))
+  }
+
+  return (
+    <Box sx={{ display: 'flex', flexDirection: 'column', alignItems: 'center', padding: 1 }}>
+      <FormControl variant='standard'>
+        <Box sx={{ display: 'flex', alignItems: 'center' }}>
+          <TooltipButton onClick={increment(-largeIncrement)} tooltipTitle={`-${largeIncrement}`} sx={{ minWidth: '30px', padding: '6px' }}><KeyboardDoubleArrowDownIcon /></TooltipButton>
+          <TooltipButton onClick={increment(-smallIncrement)} tooltipTitle={`-${smallIncrement}`} sx={{ minWidth: '30px', padding: '6px' }}><KeyboardArrowDownIcon /></TooltipButton>
+          <TextField
+            id={label}
+            type='text'
+            value={inputValue}
+            onChange={handleInputChange}
+            onFocus={(e) => { e.target.select() }}
+            onBlur={handleInputCommit}
+            onKeyDown={(e) => { if (e.key === 'Enter') handleInputCommit() }}
+            sx={{ mx: 0, width: '100px' }}
+            variant='outlined'
+            label={label}
+            InputLabelProps={{ shrink: true }}
+          />
+          <TooltipButton onClick={increment(smallIncrement)} tooltipTitle={`${smallIncrement}`} sx={{ minWidth: '30px', padding: '6px' }}><KeyboardArrowUpIcon /></TooltipButton>
+          <TooltipButton onClick={increment(largeIncrement)} tooltipTitle={`${largeIncrement}`} sx={{ minWidth: '30px', padding: '6px' }}><KeyboardDoubleArrowUpIcon /></TooltipButton>
+        </Box>
+      </FormControl>
+    </Box>
+  )
+}
+
+function ArmManualInput () {
+  const controlConfigs = [
+    { name: 'speed', label: 'Speed', min: 0, max: 5 },
+    { name: 'rotunda', label: 'Rotunda', min: -180, max: 180 },
+    { name: 'elbow', label: 'Elbow', min: -75, max: 90 },
+    { name: 'shoulder', label: 'Shoulder', min: -45, max: 90 },
+    { name: 'wristPitch', label: 'Wrist Pitch', min: -90, max: 90 },
+    { name: 'wristRoll', label: 'Wrist Roll', min: -180, max: 180 },
+    { name: 'endEffector', label: 'End Effector', min: 0, max: 90 }
+  ]
+
+  const initialControlValues = controlConfigs.reduce((acc, control) => {
+    acc[control.name] = 0
+    return acc
+  }, {})
+
+  const [controlValues, setControlValues] = useState(initialControlValues)
+
+  const handleControlChange = (controlName) => (newValue) => {
+    setControlValues(prevValues => ({
+      ...prevValues,
+      [controlName]: newValue
+    }))
+  }
+
+  const resetControlValues = () => {
+    setControlValues(initialControlValues)
+  }
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      // TODO: connect to command server, arm endpoint
+      console.log('Sending values to backend:', controlValues)
+    }, 100)
+    return () => clearInterval(interval)
+  }, [])
```

Comment:
something I've been looking into but have not tried yet is using TanStack query since it has an [interval parameter](https://tanstack.com/query/latest/docs/react/reference/useQuery) among other really nice benefits. Thought I'd share 🤷‍♂️